### PR TITLE
[benchmark] Refactor code for Benchmark test.

### PR DESCRIPTION
**Pull request by BruceDai**
This PR splits **index.js** into 
- main.js
- utils.js/utils_ic.js/utils_od.js/utils_sd.js/utils_ss.js

, and also does some optimizations.

@ibelem PTAL, thanks

### CODE REVIEW

File: benchmark/utils_sd.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    this.modelVersion = null;
+    this.outputStride = null;
+    this.scaleFactor = null;
+    this.minScore = null;
+    this.scaleWidth = null;
+    this.scaleHeight = null;
+    this.scaleInputSize = null;
+    this.heatmapTensor = null;
+    this.offsetTensor  = null;
+  }
```
Comment (ibelem) [inline benchmark/utils_sd.js]:
```
this.modelInfoDict = null;
    this.model = null;
    this.labels = null;
    this.inputTensor = null;
    this.inputSize = null;
    this.modelVersion = null;
    this.outputStride = null;
    this.scaleFactor = null;
    this.minScore = null;
    this.scaleWidth = null;
    this.scaleHeight = null;
    this.scaleInputSize = null;
    this.heatmapTensor = null;
    this.offsetTensor  = null;
```
Please reconsider if this kind of code block are necessary, also please invoke deleteAll().

```
  deleteAll() {
    if (this.model._backend != 'WebML') {
      this.model._compilation._preparedModel._deleteAll();
    }
  }
```

File: benchmark/utils_od.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    this.inputSize = null;
+    this.outputTensor = null;
+    this.outputSize = null;
+    this.outputBoxTensor = null;
+    this.outputClassScoresTensor = null;
+    this.deQuantizedOutputBoxTensor = null;
+    this.deQuantizedOutputClassScoresTensor = null;
+    this.deQuantizeParams = null;
+    this.anchors = null;
+    this.ooModelType = null;
```
Comment (ibelem) [inline benchmark/utils_od.js]:
```
this.ooModelType = null; 
this.ooModelMargin = null;
```
It's extended class, can we remove "oo" here?

File: benchmark/main.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        imageElement.src = URL.createObjectURL(inputFile);
+      } else {
+        if (modelClass === 'skeleton_detection') {
+          imageElement.src = document.getElementById('poseImage').src;
+        } else if (modelClass === 'object_detection') {
+          imageElement.src = document.getElementById('objectDetectionImage').src;
+        } else if (modelClass === 'semantic_segmentation') {
+          imageElement.src = document.getElementById('segmentationImage').src;
+        } else {
+          imageElement.src = document.getElementById('imageClassificationImage').src;
```
Comment (ibelem) [inline benchmark/main.js]: More use cases will be there, do you think the switch will be better than if-else- statement?

**Comment by BruceDai** [comment]
@ibelem @pinzhenx @Wenzhao-Xiang PTAL, thanks

**Comment by ibelem** [comment]
@BruceDai Is that possible to reuse jquery file in examples? There is code match issue for test/tools/MeasureAccuracy/resources/third_party/jquery-3.2.1.slim.min.js , how about to replace this file by reusing examples/static/lib/jquery/jquery.min.js ? Or copy examples/static/lib/jquery/jquery.min.js to test/tools/MeasureAccuracy/resources/third_party/ to use jquery.min.js in benchmark test.

**Comment by BruceDai** [comment]
@ibelem Currently benchmark test already reused jquery.min.js.  I will submit an new PR for test/tools/MeasureAccuracy by reusing jquery.min.js, thanks.

**Comment by pinzhenx** [comment]
Awesome. It looks good to me. And I hope we can do the same for utils.js in examples in the future.

**Comment by ibelem** [comment]
Great job, although more improvements needed, merge this PR at first.

**Pull request merged**